libnavigation-core/src/main/java/com/mapbox/navigation/core/reroute/MapboxRerouteController.kt

```diff
@@ -168,13 +189,13 @@ internal class MapboxRerouteController(
     @MainThread
     override fun interrupt() {
         if (state == RerouteState.FetchingRoute) {
-            val id = requestId
-            checkNotNull(id)
-            directionsSession.cancelRouteRequest(id)
-            logD(
-                "Route request interrupted",
-                LOG_CATEGORY
-            )
+            requestId?.also { id ->
```

Comment:
Why did you decide to change it? And why in this PR? It's not related. 
You also changed the logic. It looks better to me now but are you sure it has to be like this? For some reason the non-null check was introduced earlier, do you know this reason?

Reply:
Also, I think we should add a test that we don't crash if we interrupt when no request is running.

Reply:
I was investigating this last week and found the root cause of why @tomaszrybakiewicz had to add the workaround, follow up in https://github.com/mapbox/mapbox-navigation-android/pull/6683. While we should never run into a situation when the request would be `null` during a `fetching` state, I don't think that assertion is truly needed.